web/src/main/webapp/js/main-template.js

```diff
@@ -0,0 +1,591 @@
+var host;
+
+// Deployment-scripts can insert host here.
+////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+// We know that you love 'free', we love it too :)! And so our entire software stack is free and even Open Source!      
+// Our routing service is also free for certain applications or smaller volume. Be fair, grab an API key and support us:
+// https://graphhopper.com/#directions-api Misuse of API keys that you don't own is prohibited and you'll be blocked.                    
+////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+if (!host) {
+    if (location.port === '') {
+        host = location.protocol + '//' + location.hostname;
+    } else {
+        host = location.protocol + '//' + location.hostname + ":" + location.port;
+    }
+}
+
+global.d3 = require('d3');
+var L = require('leaflet');
+require('leaflet-loading');
```

Comment:
leaflet-loading inclusion part 2.

I was not able to use this npm inclusion mechanism for the other leaflet / jquery plugins. I also don't feel that this is super important at the moment to get all packages via npm.